02-microarray/differential-expression_microarray_01_2-groups.Rmd

```diff
@@ -238,43 +234,70 @@ all.equal(colnames(df), metadata$geo_accession)
+While the `genotype/variation` column contains the group information we will be using for differential expression, the `/` it contains in its column name makes it more annoying to access.  
 Accessing variable that have names with special characters like `/`, or spaces, require extra work-arounds to ignore R's normal interpretations of these characters.
+Here we will rename it as just `genotype` to make our lives later much easier.
+
+We will also recode the contents of the column, as `overexpressing the human CREB"` is a bit of an unruly name. 
+To do this, we will use the `fct_recode()` function from the `forcats` package, simplifying `"overexpressing the human CREB"` to just `CREB`.
+We will also use `fct_relevel()` to make sure our `control` samples appear first in the factor levels.
+
 
 ```{r}
+# These renaming steps will not be the same (or might not be needed at all)
+# with a different dataset
 metadata <- metadata %>%
-  dplyr::rename("genotype" = `genotype/variation`) # This step will not be the same (or might not be needed at all) with a different dataset
+  # rename the column
+  dplyr::rename("genotype" = `genotype/variation`) %>%
+  # change the names and order of the genotypes (making the column a factor)
+  dplyr::mutate(
+    genotype = genotype %>%
+      # rename the "overexpressing..." genotype to "CREB"
+      forcats::fct_recode(CREB = "overexpressing the human CREB") %>%
+      # make "control" the first level of the factor
+      forcats::fct_relevel("control")
+  )
 ```
 
 Now we will create a model matrix based on our newly renamed `genotype` variable.
 
 ```{r}
 # Create the design matrix from the genotype information
-des_mat <- model.matrix(~ metadata$genotype)
+des_mat <- model.matrix(~genotype, data = metadata)
+
+# Look at the design matrix
+head(des_mat)
 ```
 
+When we look at this design matrix, we see that there is now a `genotypeCREB` column that defines the group for each sample: 0 for control samples and 1 for the CREB samples. 
+(The model will also fit an intercept for all samples, so we can see that here as well.)
+
 ## Perform differential expression
 
-After applying our data to linear model, in this example we apply empirical Bayes smoothing and Benjamini-Hochberg multiple testing correction.
+We will use the `lmFit()` function from the `limma` package to test each gene for differential expression between the two groups using a linear model.
+After fitting our data to the linear model, in this example we apply empirical Bayes smoothing with the `eBayes()` function.
+
+Here's a [nifty article and example](http://varianceexplained.org/r/empirical_bayes_baseball/) about what the empirical Bayes smoothing is for [@bayes-estimates].
+
+Because we are testing many different genes at once, we also want to perform some multiple test corrections, which we will do with the Benjamini-Hochberg method while making a table of results with `topTable()`.
```

Comment:
Should this be right before `topTable()` and the `toptable()` bit be made into its own code chunk?

Reply:
Yeah, that makes sense, and I think it is that way in the other notebook here.